modules/@apostrophecms/asset/index.js

```diff
@@ -867,6 +870,143 @@ module.exports = {
         return process.env.APOS_ASSET_CACHE ||
               path.join(self.apos.rootDir, 'data/temp/webpack-cache');
       },
+      // Run automatically build task when appropriate
+      async autorunUiBuildTask() {
+        if (
+        // Do not automatically build the UI if we're starting from a task
+          !self.apos.isTask() &&
+            // Or if we're in production
+            process.env.NODE_ENV !== 'production' &&
+            // Or if we've set an app option to skip the auto build
+            self.apos.options.autoBuild !== false
+        ) {
+
+          checkModulesWebpackConfig(self.apos.modules, self.apos.task.getReq().t);
+          // If starting up normally, run the build task, checking if we
+          // really need to update the apos build
+          await self.apos.task.invoke('@apostrophecms/asset:build', {
+            'check-apos-build': true
+          });
+          return true;
+        }
+        return false;
+      },
+      // Start watching assets from `modules/` and
+      // every symlinked package found in `node_modules/`.
+      // `rebuildCallback` is invoked with queue length argument
+      //  on actual build attempt only.
+      // It's there mainly for testing and debugging purposes.
+      async watchUiAndRebuild(rebuildCallback) {
+        if (self.buildWatcherDisable || self.buildWatcher) {
+          return;
+        }
+        const rootDir = self.apos.rootDir;
+        // chokidar may invoke ready event multiple times,
+        // we want one "watch enabled" message.
+        let logedOnce = false;
+        const logOnce = (...msg) => {
+          if (!logedOnce) {
+            self.apos.util.log(...msg);
+            logedOnce = true;
+          }
+        };
+        const error = self.apos.util.error;
+        const _queue = [];
+        let _queueLength = 0;
+        let _queueRunning = false;
+
+        const debounceRebuild = _.debounce(chain, self.buildWatcherDebounceMs, {
+          leading: false,
+          trailing: true
+        });
+
+        const symLinkModules = await findSymlinks();
+        const watchDirs = [
+          './modules/**/ui/apos/**',
+          './modules/**/ui/src/**',
+          './modules/**/ui/public/**',
+          ...symLinkModules.reduce(
+            (prev, m) => [
+              ...prev,
+              `./node_modules/${m}/views/ui/apos/**`,
```

Comment:
what is "views" doing here? We don't use that folder for CSS and JS (only server side templates).